packages/lib-survey/src/enps.ts

```diff
@@ -0,0 +1,90 @@
+import { z } from "zod";
+
+export const SegmentedEnpsScores = z.object({
+    promoters: z.number(),
+    passives: z.number(),
+    detractors: z.number()
+});
+
+export type SegmentedEnpsScores = z.infer<typeof SegmentedEnpsScores>;
+
+/**
+ * Employee Net Promoter Score (eNPS) is a way of measuring how likely your employees are to recommend your organization as a good place to work.
+ * It asks one simple question: "On a scale of 0-10, how likely are you to recommend [company name] as a place to work?"
+ *
+ * Responses to the eNPS 0 - 10 satisfaction scale are graded into three categories:
+ *
+ * - Promoters (9 - 10): The most satisfied employees, happy and motivated
+ * - Passives   (7 - 8): These employees are content enough, but not passionate about the business
+ * - Detractors (0 - 6): Dissatisfied employees who wouldn't recommend the company
+ *
+ * To calculate your company eNPS, first remove the Passives. Subtract the percentage of Detractors from the percentage of Promoters, and round
+ * it to the nearest whole number.
+ *
+ * - %Promoters – %Detractors = eNPS
+ * - (Promoters / (Promoters + Passives + Detractors) * 100) - (Detractors / (Promoters + Passives + Detractors) * 100) = eNPS
+ * - (Promoters - Detractors) / (Promoters + Passives + Detractors) * 100 = eNPS
+ *
+ * Thus, a company with 60% Promoters and 15% Detractors gives a score of 45.
+ *
+ * eNPS scores can, theoretically, range from +100 (every employee is a Promoter and you have the perfect company) to -100 (every employee is a
+ * Detractor and your company is in trouble), with anything above 0 being an acceptable score.
+ *
+ * A good eNPS score is between 10 and 30; above 30 is excellent.
+ *
+ * @link https://delighted.com/nps-calculator
+ * @param promoters Total number of responses that were promoters
+ * @param passives Total number of responses that were passives
+ * @param detractors Total number of responses that were detractors
+ * @returns eNPS Score calculation
+ */
+export const calculate = ({
+    promoters,
+    passives,
+    detractors
+}: SegmentedEnpsScores): number =>
+    Math.floor(
```

Comment:
Potential `nan` error when the input values are all `0`. Might be good to catch that? i.e if the sum of the inputs  are 0, throw error or exit early with 0?